### ISSUE TITLE
[BUG] 0.4.0 Regression in -?>> macro

### ISSUE THREAD

**Issue by eccentric-j** [bakpakin/Fennel#276]
-?>> macro may not apply all the args on the last function.  The -?> macro seems to work as expected though.

# Reproduction
Eval the following script
```clojure
(fn identity
  [x]
  x)

(fn join
  [sep list]
  (table.concat list sep))

(print "\nTHREAD"
      (hs.inspect
        (-?>> [:a :b]
              (join " "))))
```

## Expected

Output : "a b"

## Actual

An error is thrown
```
2020-05-13 21:48:44: *** ERROR: <file>:104: bad argument #1 to 'concat' (table expected, got nil)
stack traceback:
	[C]: in function 'table.concat'
```

## Notes

If you eval the following using identity as the last form, it works:

```
(fn identity
  [x]
  x)

(fn join
  [sep list]
  (table.concat list sep))

(print "\nTHREAD"
       (hs.inspect
        (-?>> [:a :b]
              (join " ")
              (identity))))
```

### Meta
- Discovered in https://github.com/agzam/spacehammer/issues/57

**Comment by thatismatt** [comment]
This is not specific to `-?>>`, I've pinned this down to an oddity with macros that return their first argument:
```
>> (macro foo [val] val)
>> (foo (table.concat [:a :b] :c))
"ab"
>> (table.concat [:a :b] :c)
"acb"
```

**Comment by jaawerth** [comment]
Thanks, all, for so quickly noticing and fixing this! Should now be fixed in master thanks to @thatismatt's PR.

**Comment by eccentric-j** [comment]
@thatismatt Nice catch. Should have drilled a bit further but thanks for the fix!
@jaawerth I really appreciate for the fast turn around. Does that mean it's available on luarocks now?

**Comment by jaawerth** [comment]
@eccentric-j Sorry, I should have been more specific. This is now fixed in the master branch, which is usable if checked out and built with `make fennel`, and once we fix a couple of the other little regressions/bugs that have made it into 0.4.0 we'll cut a 0.4.1.

We should be able to get a 0.4.1 out in a couple days, once we have the regression bugs I filed today (and any others that crop up) fixed. I'm hesitant to rush a 0.4.1 and then miss something else and have to immediately cut a 0.4.2 but hopefully there isn't much else to miss :wink:.
